src/packaging/kpatch/commit_to_overlay.sh

```diff
@@ -0,0 +1,459 @@
+                 "'n' to delete the branch and exit,\nor 'm' to retain the branch for manual" \
+                 "push at a later time.${NO_COLOR}\n\n"
+        ;;
+    esac
+  done
+}
+
+# Cleanup artifacts
+_cleanup() {
+  cd $RUN_DIR
+  echo -en "\nCleaning up and exiting\n"
+
+  if [[ -d ${LINUX_SRC} ]] && [[ -d ${VDO_TREE} ]] && [[ ${DEBUG} != 0 ]]; then
+    if [[ ${MANUAL_PUSH} != 0 ]]; then
+      echo "Removing ${LINUX_SRC}"
+      rm -rf ${LINUX_SRC}
+    fi
+
+    echo "Removing ${VDO_TREE}"
+    rm -rf ${VDO_TREE}
+  fi
+
+  if [[ -e ${commit_file} ]] && [[ -f ${commit_file} ]]; then
+    rm -fv ${commit_file}
+  fi
+
+  unset -f CHANGE_LOG DEBUG GITDATE GITAUTHOR LINUX_SRC
+}
+
+###############################################################################
+# main()
+trap _cleanup EXIT
+test "$#" -lt 5 && print_usage
+
+process_args $@
+print_config
+checkout_source_branch
+$PROCESS_INPUT_FX
+
+echo # Intentional blank line
+
+# Store linux repo location as an absolute path in the current directory
+LINUX_SRC=${RUN_DIR}/$(mktemp -d vdo-kernel-XXXXXX)
+VDO_TREE=${RUN_DIR}/$(mktemp -d vdo-raw-XXXXXX)
+
+# Shallow clone the specified dm-linux repo branch into LINUX_SRC
+echo "Cloning kernel repo $(get_repo_name ${KERNEL_REPO_URL})/${KERNEL_BRANCH}"
+git clone --branch ${KERNEL_BRANCH} --depth 1 ${KERNEL_REPO_URL} ${LINUX_SRC}
+cd $LINUX_SRC
+git checkout -b ${OVERLAY_BRANCH} ${KERNEL_BRANCH}
+
+# Need to detect that our repo is x number of commits ahead of dm-vdo/vdo-devel
+# Clone the current repo as ${VDO_TREE}? Is VDO_TREE even needed? Aren't we working from the clone when this script is called? Need to make a copy?
+echo -en "\nCloning VDO tree $(get_repo_name ${SOURCE_REPO_URL})\n"
+git clone $SOURCE_REPO_URL ${VDO_TREE}
+
+# Loop over each commit in COMMIT_SHAS and overlay it on LINUX_SRC
+for change in ${COMMIT_SHAS[@]}; do
+  echo -en "\nProcessing changes from commit $change\n"
+
+  # Remove existing dm-vdo in case we have any deletes.
+  # XXX: The kpatch overlay process doesn't handle deletes at the moment, which
+  #      is a bug that should get handled in the future by the Makefile process
+  #      there, but it's going to get dealt with here for now.
+  if [ -d drivers/md/dm-vdo ]; then
+    rm -rf drivers/md/dm-vdo
+  fi
+
+  cd ${VDO_TREE}
+
+  # Clean up from the last build, if necessary
+  if [[ "${change}" != "${COMMIT_SHAS[0]}" ]]; then
+    echo -en "Cleaning up from the last build... "
+    git clean -fdxq
+    echo "Done"
+  fi
+
+  # Check out the new version to build and post - do we need to do this - it puts us in a detached HEAD state, so commits will not be retained
+  #sudo git config --system advice.detachedHead "false"
+  echo "Checking out the change to build and post it"
+  git checkout ${change}
+
+  # Capture the commit message from the change
+  echo "Creating the initial commit file using the commit message from the change"
+  commit_file=$(mktemp /tmp/overlay_commit_file.XXXXX)
```

Comment:
Would it make more sense to put the logfile in ${RUN_DIR} instead of in /tmp?

Reply:
It could. It is my understanding that systemd-tmpfiles applies an ageing concept to files in /tmp  and /var/tmp so that they are automatically cleaned at regular intervals. As such, there is less harm if the script somehow fails to clean up the commit files or build log.

Reply:
Having the in $RUN_DIR seems like it would make the easier to find. Does the script mention anywhere where the logs will be? That might be a reasonable compromise.

Reply:
Re-reading the script, the commit_file is temporary and we delete it on cleanup anyway, so /tmp is fine.